### ISSUE TITLE
AccName forces whitespace joiners between all inline element scenarios

### ISSUE THREAD

**Issue by cookiecrook** [w3c/accname#205]
I added the following test case in this [WPT PR](https://github.com/web-platform-tests/wpt/pull/42354/files#diff-b6feaf4b99393882d2fda38764df7f0ac8a5457b5e924158adb918ce637a9b29R169). 

The Rendered content is camel-cased `ThisIsBroken` but AccName requires it be exposed to Accessibility as `T hisI sBro ken`…

```html
<button data-expectedlabel="T hisI sBro ken" data-testname="button name from content for each child (no space, inline)" class="ex">
  <span>T</span><span>hisI</span><span>sBro</span><span>ken</span>
</button>
```

The test case is admittedly contrived, but is the ARIA Working Group sure it wants to force whitespace-joiners in all element scenarios, even when the author has removed inter-element whitespace between inline elements? If the web dev needs to style a substring of a word, there is no way for them to remove the extra spaces forced on accessibility users by the algorithm.

FWIW, the AccName algorithm requires the _OPPOSITE_ for pseudo-elements, where joining _without_ space is mandatory, but also seems somewhat arbitrary IMO. I vaguely recall the group said, 

> ~"Well, at least joining without space gives authors a way to choose space, by adding extra whitespace to the CSS content value…"

...and I understand that, but then why is it only for pseudo elements?

[Update: Intention is to align with the pseudo-element behavior, but no changes have landed in the AccName spec yet. See comments below.]

**Comment by accdc** [comment]
I'm confused, I thought years ago we specified that inline elements would be treated as they are rendered visually with no whitespace.

This matches my test using your code example at:
https://whatsock.github.io/w3c-alternative-text-computation/Editable%20Live%20Input%20AccName%20Test.html

When was camel-cased specified to be broken out with whitespace in the algorithm?

**Comment by cookiecrook** [comment]
ARIA WG discussed this morning... 

Though there has been discussion of changing the AccName spec to not force this extra whitespace for inline elements, no change has landed in the spec so far. AccName still requires this extra space. (See a variety of discussions and proposals in the [whitespace project](https://github.com/w3c/accname/projects/1#card-42288231)) 

WebKit is in line with the spec (forcing the extra space behavior as described in OP), but Gecko and Chromium appear to have updated to remove the extra space (for inline element), anticipating the spec change in favor of a better user experience despite being against the spec. Though WebKit is the only one to get the `inline-block` test right.

I'm going to commit the WPT PR with the tests, after updating the first to be inline with the new intention, rather than the spec. And I'll file the bug for WebKit to align with the others, despite the spec being out of date.

**Comment by cookiecrook** [comment]
@FYI @twilco

**Comment by cookiecrook** [comment]
https://webkit.org/b/262751

**Comment by aleventhal** [comment]
Agree that we need to ensure AccName is aligned with common sense expectation that inline elements don't require the insertion of additional whitespace.

**Comment by cookiecrook** [comment]
I'm reopening because I think this was closed accidentally without a necessary AccName change. (There was a WPT change though.)

**Comment by MelSumner** [comment]
Adding this to the whitespace deepdive. 

In this example, the accessible name calculation follows these rules:

1. No `aria-label`, `aria-labelledby`, or `title` present, so the accessible name comes from the button’s text content.
2. Each `<span>` is an inline element, so **their text content is concatenated without additional spaces.**
3. Whitespace is not inserted between inline elements unless explicitly included in the markup.

Step-by-Step Computation

- <span>T</span> → "T"
- <span>hisI</span> → "hisI"
- <span>sBro</span> → "sBro"
- <span>ken</span> → "ken"

The final accessible name calculated is the result of the concatenated spans: "ThisIsBroken" (without spaces).

**Comment by MelSumner** [comment]
BTW @cookiecrook since I can't reproduce can you confirm if this is still an issue for you?

**Comment by cookiecrook** [comment]
Yes, I think this one can be closed... If there is a reasonable solution, it's likely one of the proposed idea in:
- https://github.com/w3c/accname/issues/225
